From 5fd9168b0170441eac4ab7934b6f5184dd4d8d07 Mon Sep 17 00:00:00 2001
From: Rakesh Ravanan <quic_rravanan@quicinc.com>
Date: Tue, 27 Jun 2023 15:35:32 +0530
Subject: [PATCH] mac80211: Copied MAC to net_dev structure via dev_addr_set

Netdev warning comes if when using memcpy to copy net_dev->dev_addr.
From 5.17 version, net_dev->dev_add should be filled with dev_addr_set
to aviod netdev warning. net_dev->dev_addr MAC gets filled via
dev_addr_set().

Signed-off-by: Rakesh Ravanan <quic_rravanan@quicinc.com>
---
 net/mac80211/iface.c | 9 +++++++++
 1 file changed, 9 insertions(+)

--- a/net/mac80211/iface.c
+++ b/net/mac80211/iface.c
@@ -2379,10 +2379,20 @@
 		}
 
 		ieee80211_assign_perm_addr(local, ndev->perm_addr, type);
+    ndev->addr_len = ETH_ALEN;
 		if (is_valid_ether_addr(params->macaddr))
+#if KERNEL_VERSION(5, 17, 0) > LINUX_VERSION_CODE
 			memcpy(ndev->dev_addr, params->macaddr, ETH_ALEN);
+#else
+      dev_addr_set(ndev, params->macaddr);
+#endif
+
 		else
+#if KERNEL_VERSION(5, 17, 0) > LINUX_VERSION_CODE
 			memcpy(ndev->dev_addr, ndev->perm_addr, ETH_ALEN);
+#else
+      dev_addr_set(ndev, ndev->perm_addr);
+#endif
 		SET_NETDEV_DEV(ndev, wiphy_dev(local->hw.wiphy));
 
 		/* don't use IEEE80211_DEV_TO_SUB_IF -- it checks too much */
